### PR TITLE
[Kernel] When reading _last_checkpoint use listFrom to retrieve the file size/mtime 

### PR DESCRIPTION
**Pull request by PatrickLeahey**
### Which Delta project/connector is this regarding?
- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [x] Kernel
- [ ] Other (fill in here)

### Description
Resolves https://github.com/delta-io/delta/issues/2140

Uses FileSystemClient listFrom() method to retrieve FileStatus object with file size/mtime defined

### CODE REVIEW

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java [hunk 1]
```diff
@@ -98,9 +99,18 @@ public Optional<CheckpointMetaData> readLastCheckpointFile(TableClient tableClie
...
+                fileSystemClient.listFrom(
+                    this.lastCheckpointFilePath.toString()
+                )
+            ) {
+                while (lastCheckpointFileStatusIterator.hasNext()) {
+                    lastCheckpointFile = lastCheckpointFileStatusIterator.next();
+                }
+            }
+
             JsonHandler jsonHandler = tableClient.getJsonHandler();
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java:114]: Need to handle the case when `lastCheckpointFile` doesn't exists, basically return `Optional.empty`.

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java [hunk 2]
```diff
@@ -98,9 +99,18 @@ public Optional<CheckpointMetaData> readLastCheckpointFile(TableClient tableClie
...
+            try (CloseableIterator<FileStatus> lastCheckpointFileStatusIterator =
+                fileSystemClient.listFrom(
+                    this.lastCheckpointFilePath.toString()
+                )
+            ) {
+                while (lastCheckpointFileStatusIterator.hasNext()) {
+                    lastCheckpointFile = lastCheckpointFileStatusIterator.next();
+                }
+            }
+
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java:113]: Not directly related: can you please add ` TODO: log the exception, once https://github.com/delta-io/delta/issues/2230 is resolved` in the `catch(Exception ex)` block?

**Review by vkorukanti — CHANGES_REQUESTED** [review]
thank you @PatrickLeahey for the PR.

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java (hunk 1 above)
Comment (PatrickLeahey) [reply kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java:114]: I believe this is already handled in the catch block on line 129, but do correct me if I am wrong

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java (hunk 2 above)
Comment (PatrickLeahey) [reply kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java:113]: Added

**Comment by PatrickLeahey** [comment]
@vkorukanti I believe I have addressed requested changes

**Comment by vkorukanti** [comment]
@PatrickLeahey Thanks for this PR. Closing this due to 1) `size` or `modificationTime` not being used currently when reading the files 2) unnecessary IO call to get the file status. In future if we end of using the additional data, we can reopen the PR.

**Pull request closed without merging**